src/fparser/two/Fortran2003.py

```diff
@@ -3184,6 +3185,138 @@ def match(string):
+    '''
+    subclass_names = []
+    use_names = ['Cray_Pointer_Decl_List']
+
+    @staticmethod
+    def match(string):
+        '''Implements the matching for a Cray-pointer statement.
+
+        :param string: the reader or string to match as a Cray-pointer \
+        statement.
+        :type string: \
+        :py:class:`fparser.common.readfortran.FortranReaderBase` or \
+        `str`
+        :return: a tuple of size 2 containing a string with the name \
+        "POINTER" and a cray-pointer-decl-list, if there is a match, \
+        or `None` if there is not.
+        :rtype: (str, Cray_Pointer_Decl_List) or None
+
+        '''
+        from fparser.two.utils import EXTENSIONS
+        if not 'cray-pointer' in EXTENSIONS:
+            return None
+        return WORDClsBase.match('POINTER', Cray_Pointer_Decl_List, string,
+                                 require_cls=True)
+
+
+class Cray_Pointer_Decl(Base):  # pylint: disable=invalid-name
+    '''
+    cray-pointer-decl is ( cray-pointer-name, cray-pointee-decl )
+    '''
+    use_names = ['Cray_Pointer_Name', 'Cray_Pointee_Name', 'Cray_Pointee_Decl']
+
+    @staticmethod
+    def match(string):
+        '''Implements the matching for a Cray-pointer declaration.
+
+        :param str string: the string to match as a Cray-pointer \
+        declaration.
+        :return: None if there is no match, otherwise a tuple of size \
+        2 containing the name of the pointer as the first argument and \
+        either the name of the pointee as the second argument or a \
+        Cray-pointee declaration.
+        :rtype: None, (Name, Name) or (Name, Cray_Pointee_Decl)
+
+        '''
+        if not string:
+            return None
+        strip_string = string.strip()
+        if not strip_string:
+            return None
+        if not strip_string[0] == "(":
+            return None
+        if not strip_string[-1] == ")":
+            return None
+        strip_string_nobr = strip_string[1:-1].strip()
+        line, repmap = string_replace_map(strip_string_nobr)
+        split_list = line.split(',')
+        if len(split_list) != 2:
+            return None
+        pointer_name = repmap(split_list[0]).strip()
+        pointee_str = repmap(split_list[1]).strip()
+        if pointee_str[-1] == ")":
+            return Cray_Pointer_Name(pointer_name), \
+                Cray_Pointee_Decl(pointee_str)
+        return Cray_Pointer_Name(pointer_name), Cray_Pointee_Name(pointee_str)
+
+    def tostr(self):
+        '''
+        :return: this Cray-pointee declaration as a string
+        :rtype: str
+        :raises InternalError: if the internal items list variable is \
+        not the expected size.
+        :raises InternalError: if the first element of the internal \
+        items list is None or is empty.
+        :raises InternalError: if the second element of the internal \
+        items list is None or is empty.
+        '''
+        if not len(self.items) == 2:
```

Comment:
pylint suggests:

    if len(self.items) != 2:

Reply:
Yes that's better.